ush/python/pygfs/task/aero_emissions.py

```diff
@@ -37,48 +44,439 @@ def __init__(self, config: Dict[str, Any]) -> None:
+
+        Parameters:
+        -----------
+        workdir : Union[str, os.PathLike])
+            The working directory path.
+        current_date: (str, optional)
+            The current date for fire emissions generation.
+        forecast_dates: list, optional
+            List of forecast dates.
+        Config_dict:  Dict
+            Configuration dictionary for the task.
+
+        Returns:
+        --------
+        None
+        """
+        emistype = Config_dict.emistype
+        ratio = Config_dict.ratio
+        climfiles = np.sort(glob("{}{}".format(Config_dict.climfile_str, "*.nc")))
+        coarsen_scale = Config_dict.coarsen_scale
+        output_vars = Config_dict.output_vars
+        input_vars = Config_dict.input_vars
+
+        emission_map = {'qfed': 'qfed2.emis_*.nc4',
+                        'gbbepx': 'GBBEPx_all01GRID.emissions_v*.nc',
+                        'hfed': 'hfed.emis_*.x576_y361.*nc4'}
+
+        if emistype.lower() != 'blended':
+            try:
+                basefile = glob(emission_map[emistype.lower()])
+            except KeyError as err:
+                raise KeyError(f"FATAL ERROR: {emistype.lower()} is not a supported emission type, ABORT!")
+
+        if emistype.lower() == 'hfed':
+            AerosolEmissions.process_hfed(files=basefile, out_name=Config_dict.data_out['copy'][0][0], out_vars=output_vars, input_vars=input_vars)
+        else:
+            dset = AerosolEmissions.make_fire_emission(
+                d=current_date,
+                climos=climfiles,
+                ratio=ratio,
+                scale_climo=True,
+                coarsen_scale=coarsen_scale,
+                obsfile=basefile,
+                output_vars=output_vars,
+                input_vars=input_vars)
+
+            AerosolEmissions.write_ncf(dset, Config_dict.data_out['copy'][0][0])
+
     @staticmethod
     @logit(logger)
-    def execute(workdir: Union[str, os.PathLike], aprun_cmd: str) -> None:
-        """Run the executable (if any)
+    def process_hfed(files: list, out_name: str, out_vars: list = None, input_vars: list = None) -> None:
+        """
+        Process HFED files to generate fire emissions data.
+
+        Parameters:
+        -----------
+        files: list
+            List of HFED files to process.
+        out_name: str
+            Name of the output file to save the processed data.
+        out_vars: list, optional
+            List of output variables. Default is None.
+        input_vars: list, optional
+            List of input variables. Default is None.
+
+        Returns:
+        --------
+        None
+        """
+        vrs = out_vars
+        invars = input_vars
+
+        try:
+            if len(files) == 0:
+                raise Exception("FATAL ERROR: No HFED files were found")
+        except Exception as ee:
+            logger.exception("FATAL ERROR: HFED files not found.")
+            raise Exception(f"FATAL ERROR: Unable to find files, ABORT")
```

Comment:
I don't think this outer try is needed. No files are being read here, just the length of the list is checked. Also changed the error message since this method isn't finding files, just receiving a list.
```suggestion
        if len(files) == 0:
            raise Exception("FATAL ERROR: Received empty list of HFED files")
```